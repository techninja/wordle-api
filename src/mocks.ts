// Low DB Sync mock class.
export class MockLowSync {
  data: any = {};

  constructor(adapter: any, defaultData: any) {
    this.data = defaultData;
  }

  write(): any {}

  read(): any {}
}

// Initialize Mocks
jest.mock('lowdb', () => ({
  LowSync: jest.fn().mockImplementation(() => new MockLowSync(null, { games: [], guesses: [] })),
}));

jest.mock('lowdb/node', () => ({
  JSONFileSync: jest.fn(),
}));

jest.mock('fs', () => ({
  readFileSync: () => 'wordl',
}));
