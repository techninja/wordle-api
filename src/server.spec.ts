import './mocks';

import request from 'supertest';

import { app } from './app';

const req = request(app.callback());

describe('WordleAPI Server: Games List/Create', () => {
  test('Create new game', async () => {
    const response = await req.post('/games');
    expect(response.status).toBe(201);
    expect(response.text).toMatchSnapshot();
  });

  test('Create second new game', async () => {
    const response = await req.post('/games');
    expect(response.status).toBe(201);
    expect(response.text).toMatchSnapshot();
  });

  test('List new games', async () => {
    const response = await req.get('/games');
    expect(response.status).toBe(200);
    expect(response.text).toMatchSnapshot();
  });

  test('View game 1', async () => {
    const response = await req.get('/games/1');
    expect(response.status).toBe(200);
    expect(response.text).toMatchSnapshot();
  });

  test('View game 2', async () => {
    const response = await req.get('/games/2');
    expect(response.status).toBe(200);
    expect(response.text).toMatchSnapshot();
  });

  test('No game 3', async () => {
    const response = await req.get('/games/3');
    expect(response.status).toBe(404);
    expect(response.text).toMatchSnapshot();
  });
});

describe('WordleAPI Server: Guesses List/Create', () => {
  test('Create new guess', async () => {
    const response = await req.post('/games/1/guesses').send('testr').set('Content-Type', 'text/plain');
    expect(response.status).toBe(201);
    expect(response.text).toMatchSnapshot();
  });

  test('Create 4 more guesses', async () => {
    let response = await req.post('/games/1/guesses').send('testr').set('Content-Type', 'text/plain');
    expect(response.status).toBe(201);
    expect(response.text).toMatchSnapshot();

    response = await req.post('/games/1/guesses').send('words').set('Content-Type', 'text/plain');
    expect(response.status).toBe(201);
    expect(response.text).toMatchSnapshot();

    response = await req.post('/games/1/guesses').send('fordl').set('Content-Type', 'text/plain');
    expect(response.status).toBe(201);
    expect(response.text).toMatchSnapshot();

    response = await req.post('/games/1/guesses').send('testr').set('Content-Type', 'text/plain');
    expect(response.status).toBe(201);
    expect(response.text).toMatchSnapshot();
  });

  test('Create last fully correct guess', async () => {
    const response = await req.post('/games/1/guesses').send('wordl').set('Content-Type', 'text/plain');
    expect(response.status).toBe(201);
    expect(response.text).toMatchSnapshot();
  });

  test('Fail to create a 7th guess', async () => {
    const response = await req.post('/games/1/guesses').send('fails').set('Content-Type', 'text/plain');
    expect(response.status).toBe(422);
    expect(response.text).toMatchSnapshot();
  });

  test('View All Guesses', async () => {
    const response = await req.get('/games/1/guesses');
    expect(response.status).toBe(200);
    expect(response.text).toMatchSnapshot();
  });

  test('See no Guesses for second game', async () => {
    const response = await req.get('/games/2/guesses');
    expect(response.status).toBe(200);
    expect(response.text).toMatchSnapshot();
  });
});
