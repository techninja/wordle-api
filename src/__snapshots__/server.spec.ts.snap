// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WordleAPI Server: Games List/Create Create new game 1`] = `"{"id":1,"status":"started"}"`;

exports[`WordleAPI Server: Games List/Create Create second new game 1`] = `"{"id":2,"status":"started"}"`;

exports[`WordleAPI Server: Games List/Create List new games 1`] = `"[{"id":1,"status":"started"},{"id":2,"status":"started"}]"`;

exports[`WordleAPI Server: Games List/Create No game 3 1`] = `"Not Found"`;

exports[`WordleAPI Server: Games List/Create View game 1 1`] = `"{"id":1,"status":"started"}"`;

exports[`WordleAPI Server: Games List/Create View game 2 1`] = `"{"id":2,"status":"started"}"`;

exports[`WordleAPI Server: Guesses List/Create Create 4 more guesses 1`] = `"{"id":2,"gameId":1,"wordGuess":"testr","matches":[{"letter":"t","matchType":"none"},{"letter":"e","matchType":"none"},{"letter":"s","matchType":"none"},{"letter":"t","matchType":"none"},{"letter":"r","matchType":"partial"}]}"`;

exports[`WordleAPI Server: Guesses List/Create Create 4 more guesses 2`] = `"{"id":3,"gameId":1,"wordGuess":"words","matches":[{"letter":"w","matchType":"correct"},{"letter":"o","matchType":"correct"},{"letter":"r","matchType":"correct"},{"letter":"d","matchType":"correct"},{"letter":"s","matchType":"none"}]}"`;

exports[`WordleAPI Server: Guesses List/Create Create 4 more guesses 3`] = `"{"id":4,"gameId":1,"wordGuess":"fordl","matches":[{"letter":"f","matchType":"none"},{"letter":"o","matchType":"correct"},{"letter":"r","matchType":"correct"},{"letter":"d","matchType":"correct"},{"letter":"l","matchType":"correct"}]}"`;

exports[`WordleAPI Server: Guesses List/Create Create 4 more guesses 4`] = `"{"id":5,"gameId":1,"wordGuess":"testr","matches":[{"letter":"t","matchType":"none"},{"letter":"e","matchType":"none"},{"letter":"s","matchType":"none"},{"letter":"t","matchType":"none"},{"letter":"r","matchType":"partial"}]}"`;

exports[`WordleAPI Server: Guesses List/Create Create last fully correct guess 1`] = `"{"id":6,"gameId":1,"wordGuess":"wordl","matches":[{"letter":"w","matchType":"correct"},{"letter":"o","matchType":"correct"},{"letter":"r","matchType":"correct"},{"letter":"d","matchType":"correct"},{"letter":"l","matchType":"correct"}]}"`;

exports[`WordleAPI Server: Guesses List/Create Create new guess 1`] = `"{"id":1,"gameId":1,"wordGuess":"testr","matches":[{"letter":"t","matchType":"none"},{"letter":"e","matchType":"none"},{"letter":"s","matchType":"none"},{"letter":"t","matchType":"none"},{"letter":"r","matchType":"partial"}]}"`;

exports[`WordleAPI Server: Guesses List/Create Fail to create a 7th guess 1`] = `"Game (ID 1) has ended and is no longer accepting guesses. The word was 'wordl'."`;

exports[`WordleAPI Server: Guesses List/Create See no Guesses for second game 1`] = `"[]"`;

exports[`WordleAPI Server: Guesses List/Create View All Guesses 1`] = `"[{"id":1,"gameId":1,"wordGuess":"testr","matches":[{"letter":"t","matchType":"none"},{"letter":"e","matchType":"none"},{"letter":"s","matchType":"none"},{"letter":"t","matchType":"none"},{"letter":"r","matchType":"partial"}]},{"id":2,"gameId":1,"wordGuess":"testr","matches":[{"letter":"t","matchType":"none"},{"letter":"e","matchType":"none"},{"letter":"s","matchType":"none"},{"letter":"t","matchType":"none"},{"letter":"r","matchType":"partial"}]},{"id":3,"gameId":1,"wordGuess":"words","matches":[{"letter":"w","matchType":"correct"},{"letter":"o","matchType":"correct"},{"letter":"r","matchType":"correct"},{"letter":"d","matchType":"correct"},{"letter":"s","matchType":"none"}]},{"id":4,"gameId":1,"wordGuess":"fordl","matches":[{"letter":"f","matchType":"none"},{"letter":"o","matchType":"correct"},{"letter":"r","matchType":"correct"},{"letter":"d","matchType":"correct"},{"letter":"l","matchType":"correct"}]},{"id":5,"gameId":1,"wordGuess":"testr","matches":[{"letter":"t","matchType":"none"},{"letter":"e","matchType":"none"},{"letter":"s","matchType":"none"},{"letter":"t","matchType":"none"},{"letter":"r","matchType":"partial"}]},{"id":6,"gameId":1,"wordGuess":"wordl","matches":[{"letter":"w","matchType":"correct"},{"letter":"o","matchType":"correct"},{"letter":"r","matchType":"correct"},{"letter":"d","matchType":"correct"},{"letter":"l","matchType":"correct"}]}]"`;
